mongomock/filtering.py

```diff
@@ -408,12 +413,27 @@ def func(doc_val, search_val):
     return func
 
 
-def _type_op(doc_val, search_val):
+def _type_op(doc_val, search_val, in_array=False):
     if search_val not in TYPE_MAP:
         raise OperationFailure('%r is not a valid $type' % search_val)
     elif TYPE_MAP[search_val] is None:
         raise NotImplementedError('%s is a valid $type but not implemented' % search_val)
-    return isinstance(doc_val, TYPE_MAP[search_val])
+
+    spec = TYPE_MAP[search_val]
+    if not isinstance(spec, tuple):
+        spec = (spec,)
+
+    for t in spec:
+        if callable(t):
```

Comment:
In which case is it not callable?

Reply:
This is a result of my hasty splitting by just splitting lines without revising the content. In the next PR we also check for strings, this is why the check for callable here. I'll remove this first `if` here and add there.

Reply:
Done locally. I'll update the PR (which is #755).